IHP/DataSync/Types.hs

```diff
@@ -15,7 +19,7 @@ data DataSyncMessage
     | CreateRecordsMessage { table :: !Text, records :: ![HashMap Text Value], requestId :: !Int }
     | UpdateRecordMessage { table :: !Text, id :: !UUID, patch :: !(HashMap Text Value), requestId :: !Int }
     | DeleteRecordMessage { table :: !Text, id :: !UUID, requestId :: !Int }
-    deriving (Eq, Show)
+    deriving (Eq, Show, Generic, FromJSON)
```

Comment:
Why is `Generics` needed here?

Reply:
There were quite a few errors in the DataSync files about "no FromJSON" implementation - it appears _something_ changed with ordering of TH splices now mattering in terms of if an implementation could be found.

So this was a quick fix to get a FromJSON instance that worked -- definitely not needed. Meant to remove this before pushing!

Reply:
Ok that makes sense 👍  I try to avoid Generics wherever possible because of the compile time slowdown, therefore this directly caught my eyes :)